utils/funcexp/func_json_array_append.cpp

```diff
@@ -0,0 +1,154 @@
+#include <string_view>
+using namespace std;
+
+#include "functor_json.h"
+#include "functioncolumn.h"
+#include "constantcolumn.h"
+using namespace execplan;
+
+#include "rowgroup.h"
+using namespace rowgroup;
+
+#include "joblisttypes.h"
+using namespace joblist;
+
+#include "jsonfunchelpers.h"
+using namespace funcexp::helpers;
+
+namespace funcexp
+{
+CalpontSystemCatalog::ColType Func_json_array_append::operationType(FunctionParm& fp,
+                                                                    CalpontSystemCatalog::ColType& resultType)
+{
+  return fp[0]->data()->resultType();
+}
+
+string Func_json_array_append::getStrVal(rowgroup::Row& row, FunctionParm& fp, bool& isNull,
+                                         execplan::CalpontSystemCatalog::ColType& type)
+{
+  string tmpJs = fp[0]->data()->getStrVal(row, isNull);
+  if (isNull)
+    return "";
+
+  const CHARSET_INFO* cs = fp[0]->data()->resultType().getCharset();
+
+  json_engine_t jsEg;
```

Comment:
How heave is this json_engine_t ctor call is?

Reply:
I just asked about this function taken it as an example b/c I don't want to repeat the same question many times.

Reply:
If the ctor is heavy it might worth to consider call this ctor only once sharing the json_engine_t instance b/w calls somewhere.